### PR TITLE
[FIX] Min sdk updates

### PR DESCRIPTION
**Pull request by rksh**
We bid a fond farewell to SDK versions below 19 with this update.
- targets SDK 31, as required for new play store submission
- splits GPSStatus/GNSSStatus for maximum compat (remains big cleanup target)
- rectifies permission issues for new deployment target (try blocks in BT receiver also a cleanup target)
- fixes start-on-boot Android 10+ (per https://github.com/wiglenet/wigle-wifi-wardriving/issues/435 )

### CODE REVIEW

**Comment by rksh** [comment]
testing is shockingly reliable with that last fix in. time to test it in more places!

File: README.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 
+As of December 2022, this application supports Android SDK versions 19 (KitKat) and up. For older versions, see the [2.67 release tag](https://github.com/wiglenet/wigle-wifi-wardriving/releases/tag/2.67) to build your own copy or side-load [the compiled artifact](https://github.com/wiglenet/wigle-wifi-wardriving/blob/2.67/dist/release/wiglewifiwardriving-release.apk). 
+
 ## Features
-- View and map local RF signals
+- View and map local RF signals including WiFi, Blueooth, and Cell
 - Accumulate a database of wireless signal observations
 - Search and export observed data
 - Integrate with a [WiGLE.net account](https://wigle.net) for competition, statistics, and online aggregation and visualization
+- See your standings and accomplisments per the WiGLE.net server
```
Comment (bobzilladev) [inline README.md:14]: speling

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 2]
```diff
@@ -814,7 +817,7 @@ static void setLockScreen(Fragment fragment, boolean lockScreen) {
     }
 
     public static boolean isHighDefinition() {
-        if (Build.VERSION.SDK_INT >= 17) {
+        if (Build.VERSION.SDK_INT >= 19) {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java:820]: can this check go away entirely now?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 3]
```diff
@@ -1698,12 +1702,9 @@ public String next() {
     }
 
     public static boolean isDevMode(final Context context) {
-        if(Build.VERSION.SDK_INT == 16) {
-            return android.provider.Settings.Secure.getInt(context.getContentResolver(),
-                    android.provider.Settings.Secure.DEVELOPMENT_SETTINGS_ENABLED , 0) != 0;
-        } else if (Build.VERSION.SDK_INT >= 17) {
+        if (Build.VERSION.SDK_INT >= 19) {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java:1705]: can this check go away too?

**Review by bobzilladev — APPROVED** [review]
seems cromulent. /salute

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java (hunk 3 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: done

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: done!

File: README.md (hunk 1 above)
Comment (rksh) [reply README.md]: i'm an amazing writer tho

**Pull request merged**